lib/spack/spack/test/env.py

```diff
@@ -385,3 +378,40 @@ def test_can_add_specs_to_environment_without_specs_attribute(tmp_path, mock_pac
 
     assert len(env.user_specs) == 1
     assert env.manifest.pristine_yaml_content["spack"]["specs"] == ["a"]
+
+
+@pytest.mark.parametrize("filename", [ev.lockfile_name, "as9582g54.lock", "m3ia54s.json"])
+def test_initialize_from_lockfile(tmp_path, filename):
+    """Test that we can initialize an environment from a valid lockfile"""
+
+    init_file = tmp_path / filename
+    env_dir = tmp_path / "env_dir"
+    init_file.write_text('{ "roots": [] }\n')
+    ev.initialize_environment_dir(env_dir, init_file)
+
+    assert os.path.getsize(env_dir / ev.manifest_name) == len(ev.default_manifest_yaml())
```

Comment:
Here I would assert that the lockfile in the environment is named correctly and that its content is the same as the one we used to init.